src/css/layout/text.ts

```diff
@@ -86,8 +86,34 @@ const getRangeBounds = (context: Context, node: Text, offset: number, length: nu
     return Bounds.fromClientRect(context, createRange(node, offset, length).getBoundingClientRect());
 };
 
-const breakText = (value: string, styles: CSSParsedDeclaration): string[] => {
-    return styles.letterSpacing !== 0 ? splitGraphemes(value) : breakWords(value, styles);
+const breakText = (value: string, styles: CSSParsedDeclaration, node: Text): string[] => {
+    return styles.letterSpacing !== 0 ? splitGraphemes(value) : breakLines(value, styles, node);
+};
+
+const breakLines = (value: string, styles: CSSParsedDeclaration, node: Text): string[] => {
+    const words = breakWords(value, styles);
+    const lines: string[] = [];
+
+    let offset = 0;
+    let acc = '';
+
+    for (const word of words) {
+        const rects = createRange(node, offset, acc.length + word.length).getClientRects();
```

Comment:
This will be computing the text ClientRects twice now

Reply:
Where else is it computed? As I see, it is calculated once again in the lines 30..45 for the combined line, but here it is calculated for each word separately.

Reply:
https://github.com/niklasvh/html2canvas/blob/master/src/css/layout/text.ts#L31-L40. It either would need to persist the layout information from the words into the lines, or skip splitting the texts if they are on the same line.

This does not appear to handle [word-spacing](https://developer.mozilla.org/en-US/docs/Web/CSS/word-spacing) currently?

Reply:
It handles word-spacing. This function only decides how to split the text. Instead of word by word, it splits when a word breaks into the new line. It uses `getClientRects` so word-spacing calculation is already done.

I will check if I can reuse the rects in the above function. But I would prefer keeping the logic separate.